### DISCUSSION TITLE
Using CLI without GUI/UI

### DISCUSSION THREAD

**Discussion by PedroLSF** [mortbopet/Ripes#342] [Q&A]
Dear developers,

I am developing work that intends to use Ripes through the CLI to test third-party codes, the idea is to execute the code along with a set of inputs, but the environment in which this will be implemented does not support UI/Graphical Interface.

I saw that you have been developing the CLI recently, can you tell me if this is possible?

**Comment by mortbopet** [comment]
What exactly are you asking is possible? Yes, Ripes has a CLI mode (see https://github.com/mortbopet/Ripes/blob/master/docs/cli.md). Is your question whether or not Ripes can be built in an environment that doesn't support GUIs?

**Comment by PedroLSF** [comment]
Dear @mortbopet ,

I'm planning to explore Ripes to run my RISC-V codes and collect information such as execution time and memory usage. I plan to integrate this with an eJudge (For teaching RISC-V), but eJudge is not compatible with a graphical user interface (GUI). Therefore, my intention is to use Ripes in command line mode (CLI) to perform these tasks.

**Comment by mortbopet** [comment]
Great well, it should be possible! The CLI was initially developed to support a VSCode extension, so it should probably also fit your needs.

**Comment by PedroLSF** [comment]
Thank you very much, I will start my tests.

**Comment by PedroLSF** [comment]
Dear @mortbopet ,

I performed some tests in CLI mode, I saw that it is working perfectly on Linux, but it is not possible to make Inputs in CLI mode, correct?

I think the problem is related to this TODO:
https://github.com/mortbopet/Ripes/blob/565026f3b39c2151f3ba413e8192a7866b3e244a/src/cli/clirunner.cpp#L40-L49

I'll try to compile the code on my Linux machine and try to solve the TODO mentioned, can I?

**Reply by mortbopet** [reply]
Correct! I could imagine that we need some connection between `stdin` inputs and feeding that input into https://github.com/mortbopet/Ripes/blob/0faf41b669a93a1944707cd7d111a5e9241425fe/src/syscall/systemio.h#L458

**Reply by PedroLSF** [reply]
Got it, I'll compile the code, start investigating and try to solve it.

I'm going to clone the project, using the "Building" that is in the Readme or using docker, there it is:
```
git clone --recursive https://github.com/mortbopet/Ripes.git
cd Ripes/
cmake .
Unix:               Windows:
make                jom.exe / nmake.exe / ...
```
But I believe it's the same for Linux, maybe just swap Windows for Linux
